crypto/cipher_extra/e_aes_cbc_hmac_sha1.c

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright 2011-2016 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the OpenSSL license (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/opensslconf.h>
+
+#include <stdio.h>
+#include <string.h>
+
+
+#include <openssl/aes.h>
+#include <openssl/cipher.h>
+#include <openssl/err.h>
+#include <openssl/evp.h>
+#include <openssl/objects.h>
+#include <openssl/rand.h>
+#include <openssl/sha.h>
+#include "../fipsmodule/aes/internal.h"
+#include "internal.h"
+
+#if defined(AES_CBC_HMAC_SHA_STITCH)
+
+typedef struct {
+  AES_KEY ks;
+  // Used to compute(init, update and final) HMAC-SHA1.
+  SHA_CTX head, tail, md;
+  // In encrypt case, it's eiv_len + plaintext_len. eiv is explicit iv(required
+  // TLS 1.1+). In decrypt case, it's |EVP_AEAD_TLS1_AAD_LEN(13)|.
+  size_t payload_length;
+  union {
+    unsigned int tls_ver;
+    // In encrypt case, it's not set.
+    // In decrypt case, it stores |additional_data|.
+    // additional_data = seq_num + content_type + protocol_version +
```

Comment:
Not a standard. It comes from the code usage of the field `tls_aad` in this file. `additional_data` can be found in different RFC references(TLS 1.1, 1.2 and 1.3). One example: https://www.ietf.org/rfc/rfc5246.html#section-6.2.3.3. I didn't add the link because the code does not target one specific RFC.

Reply:
Will add these RFC links
https://www.ietf.org/rfc/rfc5246.html#section-6.2.3.2